datamodel/src/main/scala/com/rasterfoundry/granary/Model.scala

```diff
@@ -0,0 +1,30 @@
+package com.rasterfoundry.granary.datamodel
+
+import io.circe._
+import io.circe.generic.semiauto._
+
+import java.util.UUID
+
+case class Model(
+    id: UUID,
```

Comment:
I'm not sure how crazy we want to go with this, but I think with our IDs it would be a good opportunity to test out [`scala-newtype`](https://github.com/estatico/scala-newtype). Might be overkill here and I'm not sure what the fallout will be for `circe` and `doobie`, so maybe best _not_ to do it, but interested in your thoughts.

Reply:
If this causes any problems I'll bail, but I'll give it a shot https://github.com/estatico/scala-newtype/blob/master/build.sbt#L25

Reply:
Yeah so I tried it out, you can see what it looks like in 5ad4586. While it might have zero run-time overhead, it requires providing implicits for the wrapped value, and I'm having some trouble with that with tapir -- `ModelEndpoints` keeps complaining `Cannot find a codec for type: com.rasterfoundry.granary.datamodel.ModelId and media type: tap$r.MediaType.TextPlain.`, even though `implicitly[Codec[ModelId, MediaType.TextPlain, _]]` is fine in the same scope. `circe` and `doobie` were fine (you can see in `NewTypeMeta` and the implicits provided in the companion object for the `ModelId` newtype), but getting tapir to play nicely has been tricky so far

Reply:
Ah oh well, was worth a shot I think